### PR TITLE
Archiver deprecation fixes + Xcode / Cocoapods Updates

### PR DESCRIPTION
**Pull request by dodongo**
Updates LUKeychainAccess to use `NSKeyedArchiver`'s `archivedDataWithRootObject:requiringSecureCoding:error:` &`NSKeyedUnarchiver`'s `unarchivedObjectOfClass:fromData:error` on iOS11 and above.

### CODE REVIEW

File: LUKeychainAccess/NSKeyedArchiver+Additions.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSKeyedArchiver (Additions)
++ (NSData *)lu_archivedDataWithRootObject:(nonnull id)object;
```
Comment (nathanbeaumont) [inline LUKeychainAccess/NSKeyedArchiver+Additions.h]: Add a new line above and below this method declaration Donald         🙃

File: LUKeychainAccess/NSKeyedArchiver+Additions.m [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
++ (NSData *)lu_archivedDataWithRootObject:(nonnull id)object {
+  if (@available(iOS 11, *)) {
+    NSError *error;
+    NSData *result = [self archivedDataWithRootObject:object
+                                requiringSecureCoding:false
+                                                error:&error];
+    if (error) {
+      [NSException raise:@"Archiver Error" format:@"NSKeyedUnarchiver encountered %@", error];
+    }
+    return result;
```
Comment (nathanbeaumont) [inline LUKeychainAccess/NSKeyedArchiver+Additions.m]: new line between curly brace

File: LUKeychainAccess/NSKeyedUnarchiver+Additions.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSKeyedUnarchiver (Additions)
++ (nullable id)lu_unarchiveObjectWithData:(NSData *)data;
+@end
```
Comment (nathanbeaumont) [inline LUKeychainAccess/NSKeyedUnarchiver+Additions.h]: add some new lines around this method call

File: LUKeychainAccess/NSKeyedUnarchiver+Additions.m [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@implementation NSKeyedUnarchiver (Additions)
+
++ (nullable id)lu_unarchiveObjectWithData:(NSData *)data {
+  if (@available(iOS 11, *)) {
+    NSError *error;
+    id result = [self unarchivedObjectOfClass:NSObject.class fromData:data error:&error];
+    if(error) {
+      [NSException raise:@"Unarchiver Error" format:@"NSKeyedUnarchiver encountered %@", error];
+    }
+    return result;
```
Comment (nathanbeaumont) [inline LUKeychainAccess/NSKeyedUnarchiver+Additions.m]: new line here plz

File: Unit-Tests/LUKeychainAccessSpec.m [hunk 5]
```diff
@@ -1,6 +1,43 @@
 #import <LUKeychainAccess/LUKeychainAccess.h>
 #import <Kiwi/Kiwi.h>
 #import "LUTestErrorHandler.h"
+#import "NSKeyedArchiver+Additions.h"
+
+@interface NSCodingCompliantObject: NSObject<NSCoding> {}
+@property(nonatomic, readwrite) float testProperty1;
+@property(nonatomic, readwrite) NSString* testProperty2;
```
Comment (nathanbeaumont) [inline Unit-Tests/LUKeychainAccessSpec.m]:
```suggestion
@property(nonatomic, readwrite) NSString *testProperty2;
```

File: Unit-Tests/LUKeychainAccessSpec.m [hunk 6]
```diff
@@ -1,6 +1,43 @@
...
+  [coder encodeObject:@(self.testProperty1) forKey:@"testProperty1"];
+  [coder encodeObject:self.testProperty2 forKey:@"testProperty2"];
+}
+
+- (nullable instancetype)initWithCoder:(nonnull NSCoder *)decoder {
+  if (self = [super init]) {
+    self.testProperty1 = [[decoder decodeObjectForKey:@"testProperty1"] floatValue];
+    self.testProperty2 = [[decoder decodeObjectForKey:@"testProperty2"] stringValue];
+  }
+  return self;
```
Comment (nathanbeaumont) [inline Unit-Tests/LUKeychainAccessSpec.m]: new line

File: Unit-Tests/LUKeychainAccessSpec.m [hunk 7]
```diff
@@ -1,6 +1,43 @@
...
+}
+
+@end
+
+@interface NoncodableObject: NSObject {}
+@property(nonatomic, readwrite) float testProperty1;
```
Comment (nathanbeaumont) [inline Unit-Tests/LUKeychainAccessSpec.m]: new lines between @interface and @end

File: Unit-Tests/LUKeychainAccessSpec.m [hunk 8]
```diff
@@ -1,6 +1,43 @@
 #import <LUKeychainAccess/LUKeychainAccess.h>
 #import <Kiwi/Kiwi.h>
 #import "LUTestErrorHandler.h"
+#import "NSKeyedArchiver+Additions.h"
```
Comment (nathanbeaumont) [inline Unit-Tests/LUKeychainAccessSpec.m]: Can we move this test fake into its own class

**Comment by dodongo** [comment]
Thanks for the review @nathanbeaumont. The behavior of `lu_unarchiveObjectWithData` should be pretty much identical to the deprecated `unarchiveObjectWithData` method (note that it doesn't make any extra assumptions about the object conforming to `NSSecureCoding` for backwards compatibility, so I think the existing `objectForKey` test in `LUKeychainAccessSpec` should suffice.

**Pull request merged**